src/main/java/com/thealgorithms/dynamicprogramming/BuyAndSellStock.java

```diff
@@ -0,0 +1,110 @@
+package com.thealgorithms.dynamicprogramming;
+
+import java.util.Arrays;
+import java.util.Scanner;
+
+/**
+ * @author Bharath Subu (https://github.com/BharathSubu)
+ */
+
+// https://en.wikipedia.org/wiki/Dynamic_programming
+
+/*
+ * You are given an integer array prices where prices[i] is the price of a given
+ * stock on the ith day.
+ * On each day, you may decide to buy and/or sell the stock. You can only hold
+ * at most one share of the stock at any time. However, you can buy it then
+ * immediately sell it on the same day.
+ * Find and return the maximum profit you can achieve.
```

Comment:
@BharathSubu  can we buy and sell stock as many times as we want?
If that is the case , we can find maximum profit in O(n) TC and O(1) SC , using greedy approch